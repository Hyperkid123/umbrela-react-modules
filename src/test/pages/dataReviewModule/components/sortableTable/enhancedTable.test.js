import React from 'react'
import EnhancedTable from '../../../../../pages/dataReviewModule/components/sortableTable/enhancedTable';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Enhanced table component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.columnData = {}
    initialProps.footerTitle = 'foo';
    initialProps.answerCount = 33;
    initialProps.data = [{
      name: 'row',
      avg: 11,
      med: 3,
      mod: 13.5
    }, {
      name: 'row 2',
      avg: 111,
      med: 33,
      mod: 130.5
    }]
  });

  it('should render correctly', () => {
    const tree = shallow(
      <EnhancedTable {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should sort table data to asc', () => {
    const wrapper = shallow(
      <EnhancedTable {...initialProps}/>
    )
    wrapper.setState({order: 'desc'})
    wrapper.instance().handleRequestSort({}, 'name');

    console.log(wrapper.state().date);
    expect(wrapper.state().order).toEqual('asc')
  })

  it('should sort table data to desc', () => {
    const wrapper = shallow(
      <EnhancedTable {...initialProps}/>
    )
    wrapper.setState({order: 'asc'})
    wrapper.instance().handleRequestSort({}, 'name');

    console.log(wrapper.state().date);
    expect(wrapper.state().order).toEqual('desc')
  })
});
