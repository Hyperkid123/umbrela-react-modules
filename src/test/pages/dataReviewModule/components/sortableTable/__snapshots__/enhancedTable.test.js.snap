// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enhanced table component should render correctly 1`] = `
<WithStyles(Table)>
  <EnhancedTableHead
    columnData={Object {}}
    onRequestSort={[Function]}
    order="asc"
    orderBy="name"
  />
  <TableBody
    component="tbody"
  >
    <WithStyles(TableRow)
      hover={true}
      key="0"
    >
      <WithStyles(TableCell)>
        row
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        11
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        3
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        13.5
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      hover={true}
      key="1"
    >
      <WithStyles(TableCell)>
        row 2
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        111
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        33
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        numeric={true}
      >
        130.5
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </TableBody>
  <TableFooter
    component="tfoot"
  >
    <WithStyles(TableRow)>
      <WithStyles(TableCell)>
        foo
        : 
        33
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </TableFooter>
</WithStyles(Table)>
`;
